File: frontend/types/task.ts
export interface Task {
  id: number;
  title: string;
  description: string | null;
  is_done: boolean;
  created_at: string;
  updated_at: string;
}

// バックエンドと更新スキーマ(TaskUpdate)を揃える。PUTは全項目を送る必要がある
export interface TaskInput {
  title: string;
  description: string | null;
  is_done: boolean;
}

// バックエンドの共通レスポンス形式に対応
export interface ApiResponse<T> {
  success: boolean;
  data: T | null;
  message: string;
}

File: backend/app/schemas/task.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class TaskCreate(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    description: str | None = None
    is_done: bool = False


class TaskUpdate(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    description: str | None = None
    is_done: bool = False


class TaskRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    description: str | None
    is_done: bool
    created_at: datetime
    updated_at: datetime
